hermes-management/src/main/java/pl/allegro/tech/hermes/management/infrastructure/prometheus/CachingPrometheusClient.java

```diff
@@ -26,19 +38,24 @@ public CachingPrometheusClient(PrometheusClient underlyingPrometheusClient, Tick
     }
 
     @Override
-    public MonitoringMetricsContainer readMetrics(String query) {
+    public MonitoringMetricsContainer readMetrics(List<String> queries) {
```

Comment:
>       The only thing to remember about is that the key should be immutable list of queries (query is immutable by definition).

So perhaps we should copy `queries` to a immutable list to enforce it and prevent modifications from outside?

Reply:
yeah, I've not done that to avoid double copinh because the only usage is via immutable list. But definitely it is my premature optimisation :D